kyo-core/shared/src/main/scala/kyo/typeMap.scala

```diff
@@ -3,14 +3,14 @@ package kyo
 import kyo.Tag.Intersection
 import scala.collection.immutable.HashMap
 
-opaque type TypeMap[+A] = HashMap[Tag[?], Any]
+opaque type TypeMap[+A] = HashMap[Tag[Any], Any]
 extension [A](self: TypeMap[A])
 
-    private inline def fatal(using t: Tag[?]): Nothing =
+    private inline def fatal[T](using t: Tag[T]): Nothing =
         throw new RuntimeException(s"fatal: kyo.TypeMap of contents [$self] missing value of type: [${t.show}].")
```

Comment:
We can probably make this accept `Tag[Any]`, the `.show` should still be accurate, right?

Reply:
Yes, but then the callers would have to invoke `erased` to pass the tags.

Reply:
Ah, since this should rarely happen, this is probably better.

Reply:
Right, I agree.